src/notifications/common.ts

```diff
@@ -35,3 +45,66 @@ export enum NotificationPreferenceType {
 export enum NotificationPreferenceStatus {
   Muted = 'muted',
 }
+
+export const notificationPreferenceMap: Partial<
+  Record<NotificationType, NotificationPreferenceType>
+> = {
+  [NotificationType.ArticleNewComment]: NotificationPreferenceType.Post,
+  [NotificationType.CommentReply]: NotificationPreferenceType.Comment,
+  [NotificationType.SquadPostAdded]: NotificationPreferenceType.Source,
+  [NotificationType.SquadMemberJoined]: NotificationPreferenceType.Source,
+};
+
+type NotificationPreferenceUnion = NotificationPreferenceComment &
+  NotificationPreferencePost &
+  NotificationPreferenceSource;
+
+export const saveNotificationPreference = async (
+  con: DataSource | EntityManager,
+  userId: string,
+  referenceId: string,
+  notificationType: NotificationType,
+  status: NotificationPreferenceStatus,
+) => {
+  const type = notificationPreferenceMap[notificationType];
```

Comment:
Can you test this by passing null or something.
Think it might be breaking here already without coming to the validationError below.

Reply:
Shouldn't break as it is a valid key, then from the map it will return `undefined`.

Also, from GraphQL standpoint it should break as it is a required parameter.